### ISSUE TITLE
[turborepo] `1.7.0` with yarn berry doesn't work

### ISSUE THREAD

**Issue by kyun** [vercel/turbo#3280]
### What version of Turborepo are you using?

1.7.0

### What package manager are you using / does the bug impact?

Yarn v2/v3 (node_modules linker only)

### What operating system are you using?

Mac

### Describe the Bug

I've upgraded turbo version `1.6.3` -> `1.7.0`. 

I've got error like this
```
thread 'main' panicked at 'Failed to execute turbo.: Os { code: 2, kind: NotFound, message: "No such file or directory" }', crates/turborepo/src/main.rs:23:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

### Expected Behavior

It should be work well like `1.6.3`

### To Reproduce
**.yarnrc.yml**
```
nodeLinker: pnp
```

```
yarn set version berry
yarn add turbo@1.7.0
yarn dev
```

**Comment by tknickman** [comment]
Reproduced, thanks for the issue - we're taking a look.

**Comment by tknickman** [comment]
@kyun you can unblock by using `nodeLinker: node-modules`, but we're identified the issue and are looking into solutions for installing with `pnp` on our end.

**Comment by alexghr** [comment]
Hey, just to add to this: I'm seeing the same thing when using `pnpm@7.24.3` and `turbo@1.7.0`. I've tried with `node-linker=hoisted` but I still get the same error.

**Comment by tknickman** [comment]
@alexghr could you provide a reproduction? I'm not seeing any issues with `pnpm@7.24.3` locally so want to make sure I'm testing correctly.

**Comment by alexghr** [comment]
Hmm, my weird environment might also be at fault. My machine runs NixOS so ever since Turbo switched to a pre-compiled binary I can't install it directly from `npm` (see #2556 and #2293). I'm using `TURBO_BINARY_PATH` to tell the JS wrapper where the binary is and I install the binary through a Nix flake (so that I can patch it for my system).

I've pushed a branch to this test repo here https://github.com/alexghr/turbo.nix-test/tree/turbo-1.7.0+pnpm. The files are copy-pasted from the `create-turbo-app` template, to which I've added `flake.nix` so that I could install `turbo`.

Commands like `turbo --version` and `turbo --help` do work with this setup.

**Comment by tknickman** [comment]
Awesome, thanks @alexghr I'll take a look.

**Comment by teamchong** [comment]
I ran into this issue as well, as a temporary workaround. specify the `TURBO_BINARY_PATH` using [yarn bin](https://classic.yarnpkg.com/lang/en/docs/cli/bin/) can fix the issue
i.e. `yarn` to `TURBO_BINARY_PATH=$(yarn bin turbo) turbo` in `package.json`

and in case you are using [dotenv-cli](https://www.npmjs.com/package/dotenv-cli)
`dotenv -v TURBO_BINARY_PATH=$(yarn bin turbo)`

**Comment by teamchong** [comment]
<img width="473" alt="image" src="https://user-images.githubusercontent.com/25894545/212235899-bb02c40c-6724-4c3d-9ecc-1c18f4364ecc.png">

it looks like the `go-turbo` TURBO_BINARY_PATH is missing in the "[yarn bin](https://yarnpkg.com/cli/bin) turbo" folder
as a temporary workaround, I set the `TURBO_BINARY_PATH` manually in `package.json`

`"build": "TURBO_BINARY_PATH=$(echo \"console.log(require('child_process').execSync('find .yarn/unplugged/turbo-'+process.platform+'-'+process. Arch+'* -name turbo').toString())\" | node) turbo run build"`

**Comment by geauser** [comment]
I use yarn PnP in a monorepo and I have the same issue.
Using a globally installed version of `turbo` work fine for now:

```sh
npm install turbo -g
```

NB: _make sure to remove any locally installed version of `turbo`_

**Comment by geauser** [comment]
> 

I think you made a typo error, it should be + I needed to remove the `x` in `x64` so I added `.replace('x', '')`.
`
TURBO_BINARY_PATH=$(echo \"console.log(require('child_process').execSync('find .yarn/unplugged/turbo-'+process.platform+'-'+process.arch.replace('x', '')+'-* -name turbo').toString())\" | node) turbo run build
`

_Note the `process.arch` instead of `process. Arch)`_

**Comment by alexghr** [comment]
> One note, for 1.7.0 there are _two_ binaries now that need to be colocated. `turbo` and `go-turbo`

Ah that was the problem! Well, in the meantime, @joedevivo opened alexghr/turbo.nix#1 to add the new binary and everything is working perfectly with pnpm and turbo 1.7.0 :)

**Comment by kyun** [comment]
Does Turborepo support `PnP` mode officially?

**Comment by tknickman** [comment]
@kyun it does not. Officially we support `node_modules` linker only. However it looks like we made things worse in 1.7.0 so we have that on our radar.

**Comment by tknickman** [comment]
This should be fixed in `turbo@1.7.5-canary.0`

**Comment by kyun** [comment]
> This should be fixed in `turbo@1.7.5-canary.0`

I know it isn't official support yet but do you recommend using PnP mode with turbo?

**Comment by Brew-Brew** [comment]
@tknickman  Turborepo supports yarn berry with pnp mode..? I heard that it doesn't support , but it works on my repo. so i want to know official response for this!
